packages/ra-core/src/dataProvider/useQueryWithStore.ts

```diff
@@ -114,19 +115,26 @@ const useQueryWithStore = <State extends ReduxState = ReduxState>(
     dataSelector: (state: State) => any = defaultDataSelector(query),
     totalSelector: (state: State) => number = defaultTotalSelector(query),
     isDataLoaded: (data: any) => boolean = defaultIsDataLoaded
-): {
-    data?: any;
-    total?: number;
-    error?: any;
-    loading: boolean;
-    loaded: boolean;
-} => {
+): StateResult => {
```

Comment:
rename to UseQueryWithStoreValue

Reply:
It would be a breaking change, right?

Reply:
Why? The type wasn't named or exported before this PR.